generators/cypress/templates/src/test/javascript/cypress/support/keycloak-oauth2.ts.ejs

```diff
@@ -4,89 +4,91 @@
 // eslint-disable-next-line spaced-comment
 /// <reference types="cypress" />
 
-import { getAuthCodeFromLocation } from './utils';
-
-Cypress.Commands.add("keycloackLogin", (user: string) => {
-  Cypress.log({name: "Login"});
+Cypress.Commands.add('getOauth2Data', () => {
+  cy.request({
+    method: 'GET',
+    url: '/oauth2/authorization/oidc',
+    followRedirect: false,
+  }).then(response => {
+    const url = new URL(response.headers['location']);
+    const realm = url.pathname.split('/', 4)[3];
+    const clientId = url.searchParams.get('client_id');
+    const data = {
+      url,
+      realmPath: `${url.origin}/auth/realms/${realm}`,
+      realm,
+      clientId,
+    };
+    cy.wrap(data).as('oauth2Data');
+  });
+});
 
-  cy.fixture(`users/${user}`).then((userData) => {
-  const authBaseUrl = Cypress.env("auth_base_url");
-  const realm = Cypress.env("auth_realm");
-  const client_id = Cypress.env("auth_client_id");
+Cypress.Commands.add('keycloackLogin', (oauth2Data: any, user: string) => {
```

Comment:
Why do we have to add something Keycloak specific? Why can't we just click on the Logout button like we can with Protractor?

Reply:
Due to https://github.com/cypress-io/cypress/issues/944.
We must authenticate oauth2 (if the domain is not the same) using api.

Reply:
We've tried not to use anything Keycloak or Okta-specific in the past. Can this be renamed to "OIDC" or similar? And can we make sure it works with Okta? Our Protractor e2e tests work with both identity providers.

Reply:
Sure, I just didn't renamed the function, it was already there.
The one I created, I named `getOauth2Data`.

Just added this todo to https://github.com/jhipster/generator-jhipster/issues/12686.
I was going to implement it, but I forgot.